src/client.ts

```diff
@@ -0,0 +1,59 @@
+import * as input from "./input";
+import {
+    DurableClientHandler,
+    DurableClientOptions,
+    HttpDurableClientOptions,
+    TimerDurableClientOptions,
+} from "durable-functions";
+import {
+    FunctionHandler,
+    FunctionInput,
+    FunctionResult,
+    InvocationContext,
+    app as azFuncApp,
+} from "@azure/functions";
+import { DurableClient } from "./durableClient/DurableClient";
+import { getClient } from "./durableClient/getClient";
+
+export function http(functionName: string, options: HttpDurableClientOptions): void {
+    addClientInput(options);
+    azFuncApp.http(functionName, {
+        ...options,
+        handler: convertToFunctionHandler(options.handler),
+    });
+}
+
+export function timer(functionName: string, options: TimerDurableClientOptions): void {
+    addClientInput(options);
+    azFuncApp.timer(functionName, {
+        ...options,
+        handler: convertToFunctionHandler(options.handler),
+    });
+}
+
+export function generic(functionName: string, options: DurableClientOptions): void {
+    addClientInput(options);
+    azFuncApp.generic(functionName, {
+        ...options,
+        handler: convertToFunctionHandler(options.handler),
+    });
+}
+
+function addClientInput(options: Partial<DurableClientOptions>): void {
+    if (options.extraInputs && !options.extraInputs.find(isDurableClientInput)) {
```

Comment:
This isn't quite right. If I have `[input.durableClient(), input.storageBlob()]`, I think it'll fall through to the else case and overwrite with just `[input.durableClient()]`

Here's how I'd do it:
```typescript
options.extraInputs = options.extraInputs ?? [];
if (!options.extraInputs.find(isDurableClientInput)) {
    options.extraInputs.push(input.durableClient());
}
```

Reply:
Why would `[input.durableClient(), input.storageBlob()]` fall through to the else case? Wouldn't `!options.extraInputs.find(isDurableClientInput)` equal `false`?

Reply:
Brainfart moment, fixed